FITist_project/user_profiles/views.py

```diff
@@ -36,11 +39,46 @@ def get_profile(request, username=None):
     if not username:
         return redirect('profiles:user_profile', username=curr_user.username)
     user = get_object_or_404(User, username=username)
-    if request.user == user:
+    if curr_user == user:
         return render(
             request,
             'user_profiles/profile_detail.html',
             {'user': user}
         )
     else:
         return redirect('profiles:user_profile', username=curr_user.username)
+
+
+@login_required(login_url='/user_profiles/login/')
+def view_program(request, username=None):
+    curr_user = request.user
+    if not username:
+        return redirect('profiles:user_profile', username=curr_user.username)
+    user = get_object_or_404(User, username=username)
+    workouts = Workout.objects.filter(program=user.profile.program_selected)
+    exercise_groups = list()
+    exercise_blocks = list()
+    all_groups = ExerciseGroup.objects.all()
+    all_blocks = ExerciseBlock.objects.all()
+    for group in all_groups:
+        if group.workout_in.program == user.profile.program_selected:
+            exercise_groups.append(group)
+    for block in all_blocks:
+        if block.assigned_group in exercise_groups:
+            exercise_blocks.append(block)
+    print('WORKOUTS', workouts)
+    print('EXERCISE_GROUPS', exercise_groups)
+    print('EXERCISE_BLOCKS', exercise_blocks)
```

Comment:
Leaving `print` methods in committed code should probably be caught by a linter. I don't recommend it.